lib/Rx/Observable/RangeObservable.php

```diff
@@ -32,16 +31,23 @@ public function __construct($start, $count, SchedulerInterface $scheduler = null
 
         $this->start     = $start;
         $this->count     = $count;
-        $this->scheduler = $scheduler ?: new ImmediateScheduler();
+        $this->scheduler = $scheduler;
 
     }
 
-    public function subscribe(ObserverInterface $observer)
+    public function subscribe(ObserverInterface $observer, SchedulerInterface $scheduler = null)
     {
 
+        if ($this->scheduler !== null) {
+            $scheduler = $this->scheduler;
```

Comment:
I guess if we did that, we won't need the scheduler in the constructor, because more often than not a scheduler will be passed in.